assets/js/src/components/views/tab-row.js

```diff
@@ -31,20 +46,190 @@ module.exports = function( app, _meta_keys ) {
 
 		changeValue: function( evt ) {
 			var value = jQuery( evt.target ).val();
+			var type = this.model.get( 'type' );
+			var fieldType = this.model.get( 'field_type' );
 			if ( '' === value ) {
-				this.model.set( 'field_value', '' );
 				this.model.set( 'field_type', '' );
+				this.model.set( 'field_value', '' );
+				this.model.set( 'field_field', '' );
+				this.model.set( 'field_subfields', {} );
+				jQuery('#'+component+' .component-table-inner ').find('select').html("<option value=''>Unused</option>").val(""); 
 			} else {
 				this.model.set( 'field_value', value );
+				// Components - Update "Field"
+				if( "component" === type ){
+					var component = jQuery( evt.target ).closest('.component-table-wrapper').attr('id');
+					this.updateAjax_Field(component, value, false);
+				}
+				// Repeaters - Update "Field"
+				else if( "wp-type-acf" === fieldType ){
+					var id = jQuery( evt.target ).closest('td').attr('id');
+					this.updateAjax_Field(id, value, false);
+				}	
 			}
 		},
 
+		changeField: function( evt ) {
+			var value = jQuery( evt.target ).val();
+			var component = jQuery( evt.target ).closest('.component-table-wrapper').attr('id');
+			// Update Data
+			this.model.set( 'field_subfields', {} );
+			if ( '' === value ) {
+				this.model.set( 'field_field', '' );
+				jQuery('#'+component+' .component-table-inner ').find('select').html("<option value=''>Unused</option>").val(""); 
+			} else {
+				this.model.set( 'field_field', value );
+				// Update subfields
+				this.updateAjax_ComponentSubfields(component, value, false);
+			}
+		},
+
+		changeSubfield: function( evt ) {
+			var value = jQuery( evt.target ).val();
+			var index = jQuery( evt.target ).attr('data-index');
+			var subfield_data = this.model.get( 'field_subfields');
+			if(!subfield_data){ subfield_data = {}; }
+			subfield_data[index] = value;
+			this.model.set( 'field_subfields', subfield_data );
+		},
+
+		optionBuilder: function( data ) {
+			console.log('optionBuilder');
+			var options_html = "<option value=''>Unused</option>";
+			jQuery.each(data.field_data, function(i, field) {
+				// console.log('data: ',data);
+				options_html += "<option class='hidden' value='"+field.key+"' data-type='"+field.type+"'>"+field.label+"</option>";
+			}); 
+			return options_html;
+		},
+
+		/**
+		 * AJAX Update: "Field" - ACF Field group's field
+		 * 
+		 * @param {string} component - ID without the "#" of the component parent row
+		 * @param {string} field_name - Parent field name/key of the sub fields, should be a repeater
+		 * @param {object} saved_fields - OPTIONAL: Pass saved subfields if you want to set pre-existing values
+		 */
+		updateAjax_Field: function( component, field_name, saved_fields ) {
```

Comment:
Can we add more explanation to this function and the `updateAjax_ComponentSubfields` to explain what and why they are doing?